draft-ietf-quic-transport.md

```diff
@@ -1610,43 +1636,51 @@ Stream ID:
 
 : The 32-bit Stream ID of the stream being terminated.
 
-Error code:
+App Protocol Error Code:
 
-: A 32-bit error code which indicates why the stream is being closed.
+: A 32-bit application protocol error code (see {{app-error-codes}}) which
+  indicates why the stream is being closed.
 
-Final offset:
+Final Offset:
 
 : A 64-bit unsigned integer indicating the absolute byte offset of the end of
   data written on this stream by the RST_STREAM sender.
 
 
-## CONNECTION_CLOSE frame {#frame-connection-close}
+## TRANSPORT_CLOSE frame {#frame-transport-close}
 
-An endpoint sends a CONNECTION_CLOSE frame (type=0x02) to notify its peer that
-the connection is being closed.  If there are open streams that haven't been
-explicitly closed, they are implicitly closed when the connection is closed.
-The frame is as follows:
+An endpoint sends a TRANSPORT_CLOSE frame (type=0x02) to notify its peer that
+the connection is being closed.  TRANSPORT_CLOSE is used to signal errors at the
+QUIC layer, or the absence of errors (with the NO_ERROR code).
+
+If there are open streams that haven't been explicitly closed, they are
+implicitly closed when the connection is closed.
+
+The TRANSPORT_CLOSE frame is as follows:
 
 ~~~
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                        Error Code (32)                        |
+|                         Error Code (32)                       |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |   Reason Phrase Length (16)   |      [Reason Phrase (*)]    ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~
 
-The fields of a CONNECTION_CLOSE frame are as follows:
+The fields of a TRANSPORT_CLOSE frame are as follows:
 
 Error Code:
 
 : A 32-bit error code which indicates the reason for closing this connection.
+  TRANSPORT_CLOSE uses codes from the space defined in {{error-codes}};
+  APPLICATION_CLOSE uses codes from the application protocol error code space
+  ({{app-error-codes}}).
```

Comment:
I considered not doing that, but it seemed more helpful to highlight the distinction here given that I wasn't repeating the format stuff in the description of APPLICATION_CLOSE.